assets/js/script.js

```diff
@@ -0,0 +1,207 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+
+// generate the list containing the final password criteria
+function criteriaGenerator (){
+    // object containing the criteria questions confirmation
+    const confirmObj = { 
+
+    lowercase: confirm("Would you like to include lowercase letters?"),
+
+    uppercase: confirm("Would you like to include uppercase letters?"),
+
+    numbers: confirm("Would you like to include numbers?"),
+
+    specialChar: confirm("Would you like to include special characters?")
+
+  };
+  // initialize the list of criteria
+  let finalCriteria = [];
+  
+  //iterate over the object keys and values
+  for (const [key, value] of Object.entries(confirmObj)) {
+    
+    
+    //if one value confirmed add the relevant criteria to the list
+    if(key === "lowercase" && value){
+
+      finalCriteria.push("abcdefghijklmnopqrstuvwxyz");
```

Comment:
Pls, refactor this by moving the password criteria to the top of your page as constants... Pls, do not push them directly on the password array but as a variable.

Reply:
I have refactored the code. Thank you!